File: backend/app/models/ai.py
from typing import List, Optional
from pydantic import BaseModel, Field

class CopyGenerationRequest(BaseModel):
    title: str = Field(..., min_length=2, max_length=150, description="Product name", examples=["Pure Kithul Treacle"])
    keywords: str = Field(..., min_length=2, max_length=200, description="2-3 brief merchant notes", examples=["organic, low sugar, traditional tapping from Deniyaya"])
    target_audience: Optional[str] = Field("Sri Lankan households and food lovers", description="Target customer segment")

class CopyGenerationResponse(BaseModel):
    title: str
    marketing_pitch: str
    highlights: List[str] = []
    generated_by: str = "PolaLink AI"

class CategorySuggestionRequest(BaseModel):
    title: str = Field(..., min_length=2, max_length=150, examples=["Pure Cinnamon Sticks (Ceylon Alba)"])
    description: Optional[str] = Field(None, examples=["Hand-picked spicy aroma"])

class CategorySuggestionResponse(BaseModel):
    title: str
    suggested_category: str
    alternative_categories: List[str] = []
    confidence: float = 0.95

File: Rust/src/models/ai.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CopyGenerationRequest {
    pub title: String,
    pub keywords: String,
    #[serde(default = "default_target_audience")]
    pub target_audience: String,
}

fn default_target_audience() -> String {
    "Sri Lankan households and food lovers".to_string()
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CopyGenerationResponse {
    pub title: String,
    pub marketing_pitch: String,
    pub highlights: Vec<String>,
    pub generated_by: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategorySuggestionRequest {
    pub title: String,
    #[serde(default)]
    pub description: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategorySuggestionResponse {
    pub title: String,
    pub suggested_category: String,
    pub alternative_categories: Vec<String>,
    pub confidence: f64,
}
